### PR TITLE
MP OpenAPI 4.0: Implement document merging and enable FAT tests

### LINKED ISSUE #28856: Adapt existing MP OpenAPI tests for 4.0
While MP OpenAPI 4.0 now produces OpenAPI v3.1 documents, many of the existing tests should still work and we should continue to run them.

### LINKED ISSUE #28861: Implement document merging on MP OpenAPI 4.0
The code which merges OpenAPI documents needs some changes to work with the MP OpenAPI 4.0 model API.

The existing merge logic should be broadly correct, but the new spec has named pathItems and webhooks which need to be merged.

### PR DESCRIPTION
**Pull request by Azquelt**
Extend our code which merges OpenAPI documents so that it can handle the OpenAPI 3.1 documents produced by MP OpenAPI 4.0.

The merging logic is mostly the same, the changes required are:
- OpenAPI 3.1 adds webhooks which are listed by name. If the names clash, one of them must be renamed.
- OpenAPI 3.1 allows named PathItems to be listed under components. If the names clash, one of them must be renamed
- MP OpenAPI 4.0 allows arbitrary model objects to appear under a Schema. We need to change how we iterate through the children of schemas to ensure we update any references under child objects correctly if the thing they reference is renamed

Merging is done using the MP OpenAPI model API, so we need to create abstractions to cope with the cases where the API has breaking changes.

Move the merging unit tests so that we can run them for both 2.0 and 4.0. Add new unit tests for the new functionality added in 4.0.

Add support for overriding the info section from config in 4.0 (this is used by the merging FAT tests)

Run our existing FAT tests from MP OpenAPI 1.x and 2.0 against MP OpenAPI 4.0 (with the exception of tests for validation which is not yet implemented).

Fixes #28861
Fixes #28856

- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

### CODE REVIEW

**Comment by Azquelt** [comment]
#build

**Comment by Azquelt** [comment]
#build
#libby
#spawn.fullfat.buckets=com.ibm.ws.microprofile.openapi.1.1_fat_tck,com.ibm.ws.microprofile.openapi_fat,com.ibm.ws.microprofile.openapi_fat_tck,com.ibm.ws.openapi.ui_fat,io.openliberty.microprofile.openapi.2.0.internal_fat,io.openliberty.microprofile.openapi.2.0.internal_fat_tck,io.openliberty.microprofile.openapi.3.0.internal_fat_tck,io.openliberty.microprofile.openapi.3.1.internal_fat_tck,io.openliberty.microprofile.openapi.4.0.internal_fat_tck,io.openliberty.microprofile.openapi.ui.internal_fat

File: dev/io.openliberty.microprofile.openapi.4.0.internal.services/.settings/org.eclipse.jdt.ui.prefs [hunk 1]
```diff
@@ -140,13 +140,13 @@ cleanup_settings_version=2
 eclipse.preferences.version=1
 editor_save_participant_org.eclipse.jdt.ui.postsavelistener.cleanup=true
 formatter_profile=_OL Standard
-formatter_settings_version=21
+formatter_settings_version=23
 org.eclipse.jdt.ui.ignorelowercasenames=true
 org.eclipse.jdt.ui.importorder=java;javax;org;com;
-org.eclipse.jdt.ui.javadoc=false
+org.eclipse.jdt.ui.javadoc=true
```
Comment (abutch3r) [inline dev/io.openliberty.microprofile.openapi.4.0.internal.services/.settings/org.eclipse.jdt.ui.prefs:146]: Was this switch over from `false` to `true` intended?

Comment (Azquelt) [reply dev/io.openliberty.microprofile.openapi.4.0.internal.services/.settings/org.eclipse.jdt.ui.prefs:146]: Yes, despite its name, this preference controls whether all types of comment stub are created for new files, classes, methods etc.

I particularly want it on to create the copyright header in new files.

**Review by abutch3r — APPROVED** [review]
LGTM

**Pull request merged**